diy_view/src/main/java/scut/carsonho/diyview/SuperEditText.java

```diff
@@ -0,0 +1,422 @@
+            // and height information of the Element
+            // x = the starting point of the component on the X-axis of the container,
+            // y = the starting point of the component on the Y-axis of the container,
+            // width = the length of the component, height = the height of the component
+
+            // b. Did not click on the icon on the left of the status
+            // 1. Get resource ID
+            icleftunclickResId = attrs.getAttr("ic_left_unclick").isPresent()
+                    ? attrs.getAttr("ic_left_unclick").get().getIntegerValue()
+                    : ResourceTable.Media_ic_left_unclick;
+            icleftunclick = getElementFromResId(getContext(), icleftunclickResId).get();
+            icleftunclick.setBounds(leftx, lefty, leftwidth, leftheight);
+            // Initialize delete icon
+            // 1. Get resource ID
+            icdeleteResId = attrs.getAttr("ic_delete").isPresent()
+                    ? attrs.getAttr("ic_delete").get().getIntegerValue()
+                    : ResourceTable.Media_delete;
+            // 2. Get icon id based on getElementFromResId
+            icdelete = getElementFromResId(getContext(), icdeleteResId).get();
+            // 3. Set icon size
+            deletex = AttrUtils.getIntFromAttr(attrs, "delete_x", 0);
+            // 3. Set icon size
+            deletey = AttrUtils.getIntFromAttr(attrs, "delete_y", 0);
+            // 3. Set icon size
+            deletewidth = AttrUtils.getIntFromAttr(attrs, "delete_width", 60);
+            // 3. Set icon size
+            deleteheight = AttrUtils.getIntFromAttr(attrs, "delete_height", 60);
+            // 3. Set icon size
+            icdelete.setBounds(deletex, deletey, deletewidth, deleteheight);
+
+            // Set the picture on the left & right side of the EditText
+            // (only the picture on the left is available in the initial state))
+            setAroundElements(icleftunclick, null, null, null);
+            // Introduction to
+            // setAroundElements (Element left, Element top, Element right, Element bottom)
+            // Function: Set icons on the top, bottom, left, and right of
+            // EditText (equivalent to ohos:element_left="" ohos:element_right="")
+            //Remarks: The incoming Element object must have setBounds (x, y, width, height), that is,
+            // the initial position, width, and height information must be set.
+            // x: The starting point of the component on the X-axis of the container
+            // y: the starting point of the component on the Y-axis of the container
+            // width: The length of the component height: the height of the component
+            // If you don't want to display it somewhere, set it to null
+
+            // No need to set setBounds (x, y, width, height)
+            // Initialize the cursor (color & thickness)
+            // Principle: Dynamically set the cursor through the reflection mechanism
```

Comment:
We are not using the reflection mechanism to set the cursor dynamically. In HMOS ,we have a specific api in TextField for the same functionality. Please remove the comment.

Reply:
Done